apis/v1alpha2/backendtlspolicy_types.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=gateway-api,shortName=btlspolicy
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// BackendTLSPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type BackendTLSPolicy struct {
```

Comment:
I think we also need an addition somewhere on the BackendRef type that describes that when this policy is targeting a Backend matching the BackendRef, implementations SHOULD reject the backend if they can't implement the BackendTLSPolicy. We should probably also add a new Reason along with this and then transition the `SHOULD` to a `MUST` when this policy graduates to Standard channel.

https://github.com/kubernetes-sigs/gateway-api/blob/18506b70f36313bfa966e29d09971a6fe568ce18/apis/v1beta1/shared_types.go#L239-L246

Reply:
Added to the `HTTPBackendRef` struct in `v1beta1/httproute_types.go`.